src/main/java/net/geforcemods/securitycraft/entity/SentryEntity.java

```diff
@@ -165,7 +168,19 @@ else if(item == SCContent.DISGUISE_MODULE.get())
 				ItemStack module = getDisguiseModule();
 
 				if(!module.isEmpty()) //drop the old module as to not override it with the new one
-					Block.spawnAsEntity(world, getPosition(), module);
+				{
+					Block.spawnAsEntity(world, pos, module);
+
+					List<Block> blocks = ((ModuleItem)module.getItem()).getBlockAddons(module.getTag());
+
+					if(blocks.size() > 0)
+					{
+						Block block = blocks.get(0);

```

Comment:
Don't really need this variable here. You can just call `blocks.get(0)` directly in the if condition below.